src/utils/vacations-utils.ts

```diff
@@ -57,17 +54,36 @@ export function parseVacationDays(user: User): VacationDays {
  * @param vacationDays - The VacationDays object.
  * @returns An object mapping years to validated numeric vacation totals and remaining days.
  */
-export function formatVacationDaysPayload(
-  vacationDays: VacationDays
-): VacationDays {
+export function formatVacationDaysPayload(vacationDays: VacationDays): VacationDays {
   const payload: VacationDays = {};
 
   Object.entries(vacationDays).forEach(([year, data]) => {
     payload[year] = {
       total: Math.max(0, data.total ?? 0),
-      remaining: Math.max(0, data.remaining ?? 0),
+      remaining: Math.max(0, data.remaining ?? 0)
     };
   });
 
   return payload;
 }
+
+/**
+ * Validates if the vacation request uses more days than the user has.
+ * @param requestedDays Number of days the user is trying to request
+ * @param unspentDays Number of unspent vacation days the user has
+ * @returns { valid: boolean, errorMessage?: string }
+ */
+export const validateVacationRequestDays = (
+  requestedDays: number,
+  unspentDays: number
+): { valid: boolean; errorMessage?: string } => {
+  if (requestedDays > unspentDays) {
+    return {
+      valid: false,
+      errorMessage: strings.vacationRequestError.tooManyDaysRequested
+        .replace("{requestedDays}", String(requestedDays))
+        .replace("{unspentDays}", String(unspentDays))
```

Comment:
I think there is a cleaner way to pass variables into the strings without using the .replace. The curley brack syntax in the string is correct. Might be worth double checking